src/main/java/org/springframework/data/repository/config/RepositoryConfigurationSource.java

```diff
@@ -118,4 +120,22 @@
 	 * @return must not be {@literal null}.
 	 */
 	Iterable<TypeFilter> getExcludeFilters();
+
+	/**
+	 * Returns the {@link BeanNameGenerator}
+	 * @return must not be {@literal null}
+	 * @since 2.0
+	 */
+	RepositoryBeanNameGenerator getBeanNameGenerator();
```

Comment:
We shouldn't need this method anymore with `generateBeanName(…)` introduced, right? If that's the case, I'd rather drop the default implementation and move the code into the implementation base class.